.github/workflows/run-e2e-tests.yml

```diff
@@ -445,17 +441,18 @@ jobs:
         uses: actions/checkout@v4.2.1
         with:
           persist-credentials: false
+
       - name: Setup Go
         uses: actions/setup-go@v5.0.2
         with:
           go-version: "1.24.0"
           check-latest: true
+          cache: false # disable caching as this job doesn't benefit from it
```

Comment:
Do we have a test run to check this? Is it not using runs-on?

Reply:
I'm just surprised this is the case.

Reply:
> Is it not using runs-on?

These tests are not using runs-on yet. A job like this would also not be migrated to self-hosted runners due to it's simplicity. This is only generating the test matrix JSON.

Migrating it to SHRs would incur a ~20s cost to boot the runner, rather than the faster boot/queue times when using a GH runner like `ubuntu-latest`.